[package] factor out `PackageExporter._get_dependencies`

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57341 tmp intern
* #57340 [package] remove save_source_file API
* #57339 [package] remove PackageExporter.file_structure
* #57338 [package] use digraph to back dependency visualization
* #57337 [package] add simple graph data structure
* #57336 [package] make extern a dict
* **#57335 [package] factor out `PackageExporter._get_dependencies`**



Mostly refactoring. The only behavioral change is that I have eliminated
the `orig_source_file` argument to `save_source_string`. I think it
doesn't provide enough marginal value (since if you have the module name
you can get the source file anyway).

Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184/)


Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184)

suo committed:
[package] factor out `PackageExporter._get_dependencies`

Mechanical movement.

[ghstack-poisoned]

suo committed:
Update on "[package] factor out `PackageExporter._get_dependencies`"


Mechanical movement.

Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184)

[ghstack-poisoned]

SplitInfinity commented on torch/package/package_exporter.py:
```diff
@@ -213,6 +213,45 @@ def get_unique_id(self) -> str:
+                    continue
+            if self._module_exists(dep_module_name):
+                dependencies[dep_module_name] = True
+        return list(dependencies.keys())
```
Could the `dict_keys` object be returned directly? It supports multiple iteration and the caller can list-ify it if desired. All of the uses of return value in `save_source_string` should work with `dict_keys`.

I'm not sure how to annotate the key type, though.

suo replied:
I think it would be `collections.abc.KeysView` but honestly I would feel pretty silly typing that lol.

suo committed:
Update on "[package] factor out `PackageExporter._get_dependencies`"


Mechanical movement.

Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184)

[ghstack-poisoned]

suo committed:
Update on "[package] factor out `PackageExporter._get_dependencies`"





Mechanical movement.

Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184/)


Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184)

[ghstack-poisoned]

suo committed:
Update on "[package] factor out `PackageExporter._get_dependencies`"





Mechanical movement.

Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184/)


Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184)

[ghstack-poisoned]

suo committed:
Update on "[package] factor out `PackageExporter._get_dependencies`"




Mostly refactoring. The only behavioral change is that I have eliminated
the `orig_source_file` argument to `save_source_string`. I think it
doesn't provide enough marginal value (since if you have the module name
you can get the source file anyway).

Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184/)


Differential Revision: [D28114184](https://our.internmc.facebook.com/intern/diff/D28114184)

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28114184).
